src/features/page_view_event/aggregate/index.js

```diff
@@ -89,12 +89,17 @@ export class Aggregate extends AggregateBase {
     chunksForQueryString.push(param('pr', info.product))
     chunksForQueryString.push(param('af', getActivatedFeaturesFlags(this.agentIdentifier).join(',')))
 
-    if (globalScope.performance && typeof (globalScope.performance.timing) !== 'undefined') {
-      var navTimingApiData = ({
-        timing: addPT(agentRuntime.offset, globalScope.performance.timing, {}),
-        navigation: addPN(globalScope.performance.navigation, {})
-      })
-      chunksForQueryString.push(param('perf', stringify(navTimingApiData)))
+    if (globalScope.performance) {
+      try {
+        const navTimingApiData = globalScope?.performance?.getEntriesByType('navigation')?.[0]
+        const perf = ({
+          timing: addPT(agentRuntime.offset, navTimingApiData, {}),
+          navigation: addPN(navTimingApiData, {})
+        })
+        chunksForQueryString.push(param('perf', stringify(perf)))
+      } catch (err) {
+        // performance API failed for some reason
```

Comment:
For these empty catch blocks, I really think we should be doing something with the error. Either capturing and reporting an internal agent error or maybe capturing and reporting a supportability metric around certain browser API failures.

Reply:
internal error sounds good